db-scheduler-boot-starter/src/main/java/com/github/kagkarlsson/scheduler/boot/autoconfigure/DbSchedulerAutoConfiguration.java

```diff
@@ -0,0 +1,98 @@
+package com.github.kagkarlsson.scheduler.boot.autoconfigure;
+
+import com.github.kagkarlsson.scheduler.Scheduler;
+import com.github.kagkarlsson.scheduler.SchedulerBuilder;
+import com.github.kagkarlsson.scheduler.SchedulerName;
+import com.github.kagkarlsson.scheduler.boot.config.DbSchedulerCustomizer;
+import com.github.kagkarlsson.scheduler.boot.config.DbSchedulerProperties;
+import com.github.kagkarlsson.scheduler.task.Task;
+import java.util.List;
+import java.util.Objects;
+import javax.sql.DataSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.AutoConfigurationPackage;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jdbc.datasource.TransactionAwareDataSourceProxy;
+
+@Configuration
+@EnableConfigurationProperties(DbSchedulerProperties.class)
+@AutoConfigurationPackage
+@AutoConfigureAfter(DataSourceAutoConfiguration.class)
+@ConditionalOnBean(DataSource.class)
+public class DbSchedulerAutoConfiguration {
+    private static final Logger log = LoggerFactory.getLogger(DbSchedulerAutoConfiguration.class);
+
+    private final DbSchedulerProperties config;
+    private final DataSource existingDataSource;
+    private final List<Task<?>> configuredTasks;
+
+    public DbSchedulerAutoConfiguration(DbSchedulerProperties dbSchedulerProperties,
+        DataSource dataSource, List<Task<?>> configuredTasks) {
```

Comment:
Will this inject all instances of `Task` on the classpath?

Reply:
It should inject all `Task`-instances in the Spring context, or an empty list if no `Task` beans is defined.

Should we throw an exception if the list of configured `Task` beans is empty or just let the scheduler idle away?